eyeshade/lib/transaction.js

```diff
@@ -1,27 +1,98 @@
 const BigNumber = require('bignumber.js')
 const getPublisherProps = require('bat-publisher').getPublisherProps
 const uuidv5 = require('uuid/v5')
+const uuid = require('uuid')
 const {
   createdTimestamp,
   normalizeChannel
 } = require('bat-utils/lib/extras-utils')
-
 const getYoutubeChannelId = require('bat-utils/lib/youtube')
 
+const knownChains = {
+  ETH: 'ethereum',
+  BTC: 'bitcoin',
+  LTC: 'litecoin'
+}
 const SETTLEMENT_NAMESPACE = {
   'contribution': '4208cdfc-26f3-44a2-9f9d-1f6657001706',
   'referral': '7fda9071-4f0d-4fe6-b3ac-b1c484d5601a',
   'manual': 'a7cb6b9e-b0b4-4c40-85bf-27a0172d4353'
 }
+const insertTransaction = `
+INSERT INTO transactions (
+  id,
+  created_at,
+  description,
+  transaction_type,
+  document_id,
+  from_account,
+  from_account_type,
+  to_account,
+  to_account_type,
+  amount,
+  settlement_currency,
```

Comment:
I would prefer we stick with the current pattern of using queries that don't include columns we're going to set to `null` or use the default value. I think it's clearer when the query is embedded in the context as well.

Reply:
does that make sense? we will have 8 queries in this one file that do almost exactly the same thing. i would think that repeating similar queries takes away from readability. would putting the insertion step into a function be preferable?